### DISCUSSION TITLE
Allow using `useBlocker` without setting a `beforeunload` listener

### DISCUSSION THREAD

**Discussion by duniul** [TanStack/router#2263] [Ideas]
Currently, `useBlocker` will always add a `beforeunload` event listener, which triggers a blocking alert box when leaving the page or closing the tab. However, there are situations where you might want to block internal navigation but don't want to block unload.

This would align it with `react-router`'s behavior, which recommends users to set their own `beforeunload` listener instead.

Maybe just a prop like this could make sense to make it backwards compatible?

```tsx
const blocker = useBlocker({ blockBeforeUnload: false })
```

Looking at the source code it would need to pass the option to the `history`'s block method, which currently only accepts functions, but I don't know if that also needs to remain backwards compatible. In that case maybe it could be added as a second argument?

---

**Edit**: Seems like https://github.com/TanStack/router/pull/1790 might solve this issue, adding `disableBeforeUnload` as an option 👍

**Comment by CanRau** [comment]
Might've been changes, didn't look too deep but it's now `enableBeforeUnload: false`
